src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java

```diff
@@ -59,100 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
-
-                            String answer = get_answers(full_response.get("answer").toString());
-                            interpretation.getFeatures().put(Linguistics.UNDERSPECIFIED_ANSWER, answer);
-                            interpretation.answer = answer;
-
-                            interpretation.getFeatures().put(Linguistics.PARSER_RESULT, Linguistics.PARSER_OUTCOME.UNDERSPECIFIED);
-                            interpretation.parserOutcome = Linguistics.PARSER_OUTCOME.UNDERSPECIFIED;
-                        }
-                        // Read tokens
-                        if (full_response.containsKey("tokens")) {
-                            interpretation.getFeatures().put(Linguistics.TOKENS, full_response.get("tokens").toString().split(","));
-                        }
-                        // Read extracted non-semantic relations
-                        if (full_response.containsKey("relations")) {
-                            interpretation.getFeatures().put(Linguistics.TRIPLE, extract_relations((Map<String, Double>) full_response.get("relations")));
-                        }
-                        // Read extracted sentiment
-                        if (full_response.containsKey("sentiment")) {
-                            interpretation.getFeatures().put(Linguistics.SENTIMENT, full_response.get("sentiment").toString());
-                        }
-                        // Read POS-tags
-                        if (full_response.containsKey("postags")) {
-                            interpretation.getFeatures().put(Linguistics.POSTAGS, full_response.get("postags").toString().split(","));
-                        }
-                        // Read lemmatized tokens
-                        if (full_response.containsKey("lemma_tokens")) {
-                            interpretation.getFeatures().put(Linguistics.LEMMAS, full_response.get("lemma_tokens").toString().split(","));
-                        }
-                        // Read utterance type
-                        if (full_response.containsKey("type")) {
-                            interpretation.getFeatures().put(Linguistics.UTTERANCE_TYPE, full_response.get("type").toString());
-                        }
-                    } catch (Exception e) {
-                        logger.error("Exception while parsing semantic response: " + e.getMessage());
-                        e.printStackTrace();
-                    }
-                }
-                return interpretation;
-            } catch (IOException e) {
-                e.printStackTrace();
-                return interpretation;
-            }
-        } else
-            return interpretation;
+        String response;
+        SemanticAnalyzerInterface.Result result = semanticAnalyzer.analyze(
+            (String) interpretation.getFeature("sentence"));
+        interpretation.getFeatures().put(Linguistics.TOKENS, result.getTokens());
```

Comment:
so why are we not handling "(no answer)" responses?
why are you ignoring parser results like followupq?